pkg/agent/core/ngt/service/ngt.go

```diff
@@ -1615,6 +1753,11 @@ func (n *ngt) IsIndexing() bool {
 	return i && ok
 }
 
+func (n *ngt) IsFlushing() bool {
+	i, ok := n.flushing.Load().(bool)
```

Comment:
it would be better to use atomic.Bool

Reply:
I've implemented the change, which has simplified the code by removing the need for type assertions when loading the boolean value. 
 https://github.com/vdaas/vald/pull/2273/commits/bc2fe50c40cfecaee1b7aca1edef7847e5e58b94